File: src/SkyBlock/enchants/armor/BaseArmorEnchant.php
<?php


namespace SkyBlock\enchants\armor;


use pocketmine\player\Player;
use SkyBlock\enchants\BaseEnchantment;

abstract class BaseArmorEnchant extends BaseEnchantment {
    /**
     * @param Player $holder
     *
     * @return bool
     */
    public function isApplicableTo(Player $holder) : bool {
        return mt_rand(1, 25) === 1;
    }

    /**
     * Called after damaging the entity to apply any post damage effects to the target.
     *
     * @param Player $attacker
     * @param Player $victim
     * @param int    $enchantmentLevel
     */
    abstract public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel) : void;
}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/BaseArmorEnchant.java
package dev.skyblock.enchants;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
public abstract class BaseArmorEnchant extends BaseEnchant {
    public BaseArmorEnchant(int id, String name, int maxLevel) { super(id, name, maxLevel); }
    public abstract void onActivation(Player player, EntityDamageByEntityEvent ev, int level);
    public abstract boolean isApplicableTo(Player player, int level);
}
